Source/ASCollectionNode.mm

```diff
@@ -682,24 +683,34 @@ - (void)waitUntilAllUpdatesAreCommitted
 - (void)reloadDataWithCompletion:(void (^)())completion
 {
   ASDisplayNodeAssertMainThread();
-  if (self.nodeLoaded) {
-    [self.view reloadDataWithCompletion:completion];
+  if (!self.nodeLoaded) {
+    return;
   }
+  
+  [self performBatchUpdates:^{
+    [self.view.changeSet reloadData];
+  } completion:^(BOOL finished){
+    if (completion) {
+      completion();
+    }
+  }];
 }
 
 - (void)reloadData
 {
   [self reloadDataWithCompletion:nil];
 }
 
-- (void)relayoutItems
+- (void)reloadDataImmediately
 {
-  [self.view relayoutItems];
+  ASDisplayNodeAssertMainThread();
+  [self reloadData];
+  [self waitUntilAllUpdatesAreCommitted];
 }
 
-- (void)reloadDataImmediately
+- (void)relayoutItems
 {
-  [self.view reloadDataImmediately];
+  [self.view relayoutItems];
```

Comment:
I know the original code doesn't assert main thread and check if this node is loaded, should we do these now?